### PR TITLE
#6 Implemented user creation flow

### PR DESCRIPTION
**Pull request by tdoff**
added implementation of user creation flow

### CODE REVIEW

File: src/main/java/kz/toko/app/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+
+    private final UserRepository repository;
+    private final UserMapper mapper;
+
+    @Override
+    public User save(CreateUserRequest body) {
+        UserEntity user = mapper.toEntity(body);
+        repository.save(user);
```
Comment (Dauren-Delmukhambetov) [inline src/main/java/kz/toko/app/service/impl/UserServiceImpl.java:25]: Add setting creation date

File: src/main/resources/db/migration/V202105292030__create_user.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
+CREATE TABLE users(
+    id bigint GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    username varchar not null,
```
Comment (Dauren-Delmukhambetov) [inline src/main/resources/db/migration/V202105292030__create_user.sql:3]: Why not use `text`?

File: src/main/java/kz/toko/app/entity/UserEntity.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import javax.validation.constraints.NotEmpty;
+
+@Data
+@Entity
+@Table(name = "users")
+public class UserEntity extends AuditableEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "sequenceGenerator")
+    @SequenceGenerator(name = "sequenceGenerator")
```
Comment (Dauren-Delmukhambetov) [inline src/main/java/kz/toko/app/entity/UserEntity.java:21]: Either rename sequence, or use `Identity` as I did

**Review by Dauren-Delmukhambetov — CHANGES_REQUESTED** [review]

File: src/main/resources/db/migration/V202105292030__create_user.sql (hunk 2 above)
Comment (tdoff) [reply src/main/resources/db/migration/V202105292030__create_user.sql]: change type of username to text

File: src/main/java/kz/toko/app/entity/UserEntity.java (hunk 3 above)
Comment (tdoff) [reply src/main/java/kz/toko/app/entity/UserEntity.java]: changed to Identity

**Review by Dauren-Delmukhambetov — APPROVED** [review]
Well done!

**Pull request merged**